models/domains_domains.php

```diff
@@ -300,8 +300,11 @@ public function renewDomain($service_id, $years = 1)
         // Determine whether invoices for this service remain unpaid
         $unpaid_invoices = $this->Invoices->getAllWithService($service->id, $service->client_id, 'open');
 
-        // Disallow renew if the current service has not been paid
-        if (!empty($unpaid_invoices)) {
+        // Determine if this domain has pending service changes
+        $service_changes = $this->ServiceChanges->getAll('pending', $service->id);
```

Comment:
I don't think this is sufficient.  In the 5 minute window between cron runs, a service may be ordered, paid for, and manually renewed, thus triggering the bug.  This kind of service would not have a ServiceChange associated with it.    This same kind of situation would also apply to renewing services that just had their invoices paid for.  So we need to check for those kind of services